### PR TITLE
Error if DB downgrades are required

### LINKED ISSUE #27906: Warn that a Liquibase rollback may need to be run when downgrading Metabase versions
**Is your feature request related to a problem? Please describe.**
If a user runs Metabase on one version and then wants to downgrade to the previous version (without relying on a backup), they should first run a command to rollback schema migrations per the instructions in https://github.com/metabase/metabase/pull/26429. But it might not be obvious that this step needs to happen.

**Describe the solution you'd like**
We should be able to automatically detect that a downgrade has happened and print a big warning about this on startup.

**Describe alternatives you've considered**
Just put it in the docs

### PR DESCRIPTION
**Pull request by johnswanson**
Before running migrations, check the major versions of:

- the latest database migration that was actually applied (e.g. if v49 ran migrations, this would be 49)

- the latest database migration that liquibase is aware of (e.g. if we're currently running v48, this would be 48)

If the database has executed migrations that we don't know about, throw an exception and exit with a helpful error message telling the user they need to run `migrate down` from the newer version before they can downgrade.

Fixes https://github.com/metabase/metabase/issues/27906

### CODE REVIEW

**Review by noahmoss — APPROVED** [review]
Functionality lgtm. 

My one question is about the error message — I wonder if we should link to the [relevant page in the docs](https://www.metabase.com/docs/latest/installation-and-operation/upgrading-metabase#rolling-back-an-upgrade).

Might be a question for @jeff-bruemmer and/or the success team. I think we have a couple other places where we have hard-coded docs links in the codebase, but not sure if that's encouraged.

**Comment by jeff-bruemmer** [comment]
> My one question is about the error message — I wonder if we should link to the [relevant page in the docs](https://www.metabase.com/docs/latest/installation-and-operation/upgrading-metabase#rolling-back-an-upgrade).

It's okay (and can be better) to link to documentation. One consideration, however, is embedding/white-labeling, but this particular case is admin territory, where white-labeling doesn't apply.

File: src/metabase/db/setup.clj [hunk 1]
```diff
@@ -53,6 +53,7 @@
                    "\n"))
     (throw (Exception. (trs "Database requires manual upgrade.")))))
 
+
```
[the remark this answers is not in the document]
Comment (johnswanson) [reply src/metabase/db/setup.clj]: Apparently this is a habit of mine 😂

**Comment by johnswanson** [comment]
> My one question is about the error message — I wonder if we should link to the [relevant page in the docs](https://www.metabase.com/docs/latest/installation-and-operation/upgrading-metabase#rolling-back-an-upgrade).

@noahmoss does this look good? Specifically, I added the link to the error message emitted just *before* the error itself is thrown - just wanted to check that that makes sense to you too.

**Comment by noahmoss** [comment]
@johnswanson Looks good!

**Pull request merged**